### PR TITLE
Mexprfix

### PR DESCRIPTION
**Pull request by david-broman**
Refactored and added a number of things to MExpr in boot:

- Added matching of sequences in Mexpr.
- Refactored to have separate pattern wildcards.
- Removed CSeq from boot.
- Removed TmIf in boot. Replaced with match.

### CODE REVIEW

File: src/boot/ast.ml [hunk 1]
```diff
@@ -152,21 +148,39 @@ and tm =
...
+| TmFix     of info                                                 (* Fix point *)
+
+
 
 and label =
 | LabIdx of int                                   (* Tuple index *)
 | LabStr of ustring                               (* Record label *)
 
+(* Kind of pattern name *)
+and name =
```
Comment (elegios) [inline src/boot/ast.ml]: I'd probably prefer pat_name or something like it, `name` seems a bit generic

File: src/boot/mexpr.ml [hunk 2]
```diff
@@ -429,7 +428,15 @@ let rec debruijn env t =
      | VarTm(y)::ee -> if y =. x then n else find fi ee (n+1) x
      | [] -> raise_error fi ("Unknown variable '" ^ Ustring.to_utf8 x ^ "'")) in
   let rec dbPat env = function
-    | PatNamed(fi,x) -> (VarTm(x)::env, PatNamed(fi,x))
+    | PatNamed(_,NameStr(x)) as pat -> (VarTm(x)::env, pat)
+    | PatNamed(_,NameWildcard) as pat -> (VarTm(us"")::env, pat)
```
Comment (elegios) [inline src/boot/mexpr.ml]: I suspect that it should be possible to not add anything to the environment here, as long as nothing is added in `tryMatch` either.

File: src/boot/ast.ml (hunk 1 above)
Comment (david-broman) [reply src/boot/ast.ml]: Sure, I changed to `patName`. We have decided to not to use snake case.

File: src/boot/mexpr.ml (hunk 2 above)
Comment (david-broman) [reply src/boot/mexpr.ml]: Thanks. I have updated to not bind the name when it is a wildcard.

**Pull request merged**